### ISSUE TITLE
Does `rls-analysis` support everything we need?

### ISSUE THREAD

**Issue by steveklabnik** [steveklabnik/rustdoc#7]
Upstream wants to work with us, but we haven't actually determined if we need anything yet.

This quest issue is both broad and specific. Someone wanting to close it out should:

1. Work through all of the different kinds of information that we need out of `rls-analysis`.
2. Inspect its API to figure out if it all exists or not.

If we don't have all of the stuff we need, we'll have to open issues with upstream and work with them to get the information landed.

**Comment by dzamlo** [comment]
After a quick look at generated JSON, it looks like the exported macros (and their doc comments) are not included.

But I'm not familiar enough with the `save-analysis` part of rust to be sure if this is a mistake on my part or if this is really not included.

**Comment by euclio** [comment]
@steveklabnik It seems that rls-analysis does not export the visibility of items? We shouldn't be documenting private items :)

**Comment by steveklabnik** [comment]
yeah, we should at least know if something is public or private; sounds like an important thing to sort out

**Comment by theindigamer** [comment]
@euclio , it would be nice to be able to document private definitions too. Earlier discussion https://github.com/rust-lang/rust/issues/15347

**Comment by mgattozzi** [comment]
We ran into some issues implementing the DefKinds. The discussion was [here](https://github.com/steveklabnik/rustdoc/pull/177#discussion_r141922958). Essentially, rls-analysis still needs to be expanded for other items so we can implement everything.

**Comment by euclio** [comment]
We will not be able to support `pub use` and reexports until https://github.com/nrc/rls-analysis/issues/123 is fixed.
